juju/juju
# [JUJU-2286] Add initial secret backend CLI - add and list

Commit 1:
Add initial secret backend CLI - add and list
Includes tweaks to facade params to include secret count in the lsit results Secret providers now properly use model config for secret-backend - they use the named provider (this needed some refactoring of the providers)
Model import also needs to account for the mandatory secret-backend model config which defaults to auto auto means use internal (controller) backend for vm models and k8s backend for caas models

Commit 2:
Driveby rename files in secret/provider to split backend and provider code.

Commit 3:
Driveby rename of old secret store method.

## Checklist


## QA steps

start a vault service, create a config.yaml file with endpoint, token, and unseal key

bootstrap a lxd controller
juju add-secret-backend myvault vault --file /path/to/config.yaml
juju add-model vaultsecrets --config secret-backend=myvault
juju deploy ubuntu
juju exec --unit ubuntu/0 "secret-add foo=bar"

check that the secret is stored in vault

add another model and do not set the secret backend
juju model-config secret-backend -> auto
add a secret and see that it is stored in the controller

juju secret-backends
```
Name      Type        Secrets
internal  controller  666  
myvault   vault       666 
```

bootstrap a k8s model
add a secret and see that the secret is stored in k8s
